inference-engine/src/vpu/graph_transformer/src/middleend/passes/allocate_resources.cpp

```diff
@@ -224,6 +224,21 @@ AllocationResult runAllocator(const Model& model, EnableShapeAllocation enableSh
                 allocateShape(tempBuffer);
             }
         }
+
+        // Allocate shape for unused inputs
+        DataVector unusedInputs;
+        std::for_each(model->datas().begin(), model->datas().end(), [&unusedInputs](const Data& data) {
```

Comment:
I guess you can use [`std::copy_if`](https://en.cppreference.com/w/cpp/algorithm/copy) + [`std::back_inserter`](https://en.cppreference.com/w/cpp/iterator/back_inserter)

```suggestion
        const auto& dataObjects = model->datas();
        std::copy_if(dataObjects.cbegin(), dataObjects.cend(), std::back_inserter(unusedInputs), [](const Data& data) { return data->usage() == DataUsage::Input && !data->isConsumed(); });
```

Reply:
Done.